packages/syft/src/syft/core/tensor/autodp/gamma_tensor.py

```diff
@@ -1730,6 +1795,29 @@ def _sum(state: dict) -> jax.numpy.DeviceArray:
             state=output_state,
         )
 
+    def mean(
+        self, axis: Optional[Union[int, Tuple[int, ...]]] = None, **kwargs: Any
+    ) -> GammaTensor:
+        def _mean(state: dict) -> jax.numpy.DeviceArray:
+            return jnp.mean(self.run(state).child)  # type: ignore
+
+        output_state = dict()
+        output_state[self.id] = self
+
+        child = self.child.mean(axis=axis, **kwargs)
+
+        min_v = child.min()
```

Comment:
Avoid directly computing min or max on the result- since this leaks information without spending privacy budget
Make use of the fact that we know the min_vals and max_vals _before_ the mean method is applied, and those numbers are stored in `self.min_vals.data` and `self.max_vals.data`

Reply:
Everything else is spot on though- in particular, well done with the `_mean()` method!

Reply:
Nice Explanation @IshanMi